packages/protocol/thread/thread.ts

```diff
@@ -205,13 +210,22 @@ class _ThreadFront {
     });
   }
 
-  get currentPause(): Pause | null {
-    return this._currentPause;
+  get currentPause(): Pause {
+    return {
+      point: this.currentPoint,
+      time: this.currentTime,
+      pauseId:
+        this.currentPauseId ??
+        getPauseIdForExecutionPointIfCached(this.currentPoint)?.value ??
+        null,
+    };
```

Comment:
This seems like an unexpected change to me. Was it done so that we could remove the `ensurePause` waiter? I'm not sure I know what the potential effects of this change are.

Reply:
To simplify the refactor, I'm returning an object containing `point`, `time` and `pauseId` as a replacement for the `Pause`. `Pause` contained a `pauseId` that may initially be set to `null` and then to the `pauseId` once the pause was created. Now we have that `pauseId` in the suspense cache, so pulling it from there should have the same effect.

Reply:
I guess if you've tested it thoroughly enough that you're confident 😁